### PR TITLE
Drup8 806

### PR DESCRIPTION
**Pull request by DeveloperPiatt**
### Description
Replace the OSU logo with a crest only version for mobile devices. When a user clicks the crest the OSU Menu should be displayed. While the menu is open display the full logo.

Mockups in Figma.

### Initial Ticket Specs
- [x] Replace the OSU Logo with a crest for sizes UNDER 1024
- [x] When user clicks crest, menu should open
- [x] Menu needs to display the full logo
- [x] Finished product should match the figma as closely as possible

### Styling Issues
- [x] Menu icon needs to be wider
- [x] Investigate possibly locking the width of the first grid column
- [x] Issue with the crest padding? Not sure but having a heck of a time with aligning things
- [x] Issue with the borders. Need help properly getting this to look correctly. Took a lot of stabs, couldn't get it right
- [x] When menu is open the close button icon isn't sized properly
- [x] The height of the title under 1024 needs to match when its over 1024
- [x] Fix the padding on the mobile site title
- [x] Vertical line doesn't go top to bottom like it does for the menu and in the mockup
- [x] Title text should be 14px for mobile (keep current size for desktop)

- [x] Close icon needs to be a button and pushed to the right
- [x] Logo (when clicked in mobile from the crest) needs some padding on the left (Stay in the same spot as before you click on it)
- [x] Desktop toggle of the menu is now broken (removes the close button, and spacing is odd)
- [ ] Crest on it's own looks smaller than crest w/ the OSU text

### Attachments

default-mobile-view (1) osu-menu

### CODE REVIEW

**Comment by DeveloperPiatt** [comment]
The crest padding issue has to do with the first grid column resizing i think. Will investigate.

**Comment by DeveloperPiatt** [comment]
Alright so I did the best I could, theres some slight mismatches from figma doc when it comes to the font awesome icons not matching because michael kept using the thin ones.

File: templates/navigation/menu--group-menu.html.twig [hunk 1]
```diff
@@ -15,17 +15,14 @@
  *     menu tree.
  *   - is_collapsed: TRUE if the link has children within the current menu tree
  *     that are not currently visible.
- *   - in_active_trail: TRUE if the link is in the active trail.
```
Comment (emerham) [inline templates/navigation/menu--group-menu.html.twig:18]: This is just a comment saying that the variable is there and what it does, probably shouldn't remove.

Comment (DeveloperPiatt) [reply templates/navigation/menu--group-menu.html.twig:18]: Noted and added back in, thanks for catching that.

File: templates/block/block--system-branding-block.html.twig [hunk 2]
```diff
@@ -103,30 +106,45 @@
...
         <img class="tw-self-center tw-max-w-none tw-h-20" src="{{ site_logo }}" alt="{{ 'Home'|t }}"/>
       </a>
-      <div class="tw-flex tw-flex-col tw-justify-self-center tw-justify-center tw-items-center lg-tw-hidden">
-        <div class="tw-cursor-pointer tw-justify-self-center tw-flex tw-flex-col tw-justify-center tw-items-center" onclick="">
-          <i class="fas fa-bars"></i>
-          <span>Menu</span>
-        </div>
-      </div>
+
+      <a href="#" title="{{ 'Home'|t }}" rel="home" class="site-logo modal-open tw-self-stretch tw-flex tw-no-underline lg-tw-hidden tw- tw-border-r-2 tw-border-neutral-200 tw-justify-center">
```
Comment (emerham) [inline templates/block/block--system-branding-block.html.twig]: extra tw- that doesn't apply to anything

File: templates/block/block--system-branding-block.html.twig [hunk 3]
```diff
@@ -11,32 +11,35 @@
...
 #}
 {% block content %}
   <!--Modal-->
   <div class="modal-overlay tw-pointer-events-auto tw-hidden tw-absolute tw-w-screen tw-z-20 tw-h-screen tw-bg-gray-900 tw-opacity-50"></div>
 
   <div
-    id="osu-menu" class="modal tw-pointer-events-auto tw-hidden tw-absolute tw-h-screen tw-overflow-y-auto tw-z-20 tw-bg-neutral-100 tw-w-screen md-tw-w-81.2">
+    id="osu-menu" class="modal tw-pointer-events-auto tw-hidden tw-absolute tw-h-screen tw-overflow-y-auto tw-z-20 tw-bg-neutral-100 tw-w-screen lg-tw-w-81.2">
     <!-- This is almost identical to chunk below, but there are a few differences that make it hard to extract into 1 file -->
     <div class="tw-grid tw-grid-cols-8 lg-tw-flex tw-flex-wrap tw-items-center lg-tw-flex-nowrap tw-pt-1 tw-bg-white">
```
Comment (emerham) [inline templates/block/block--system-branding-block.html.twig]: In 1023 and under there's an odd jump of screen height, adding a tw-h-24 here seems to fix it and I do not see any issues in desktop with this small adjustment

Comment (emerham) [reply templates/block/block--system-branding-block.html.twig]: line 24 specifically

File: templates/block/block--system-branding-block.html.twig [hunk 4]
```diff
@@ -103,30 +106,45 @@
...
       </a>
-      <div class="tw-flex tw-flex-col tw-justify-self-center tw-justify-center tw-items-center lg-tw-hidden">
-        <div class="tw-cursor-pointer tw-justify-self-center tw-flex tw-flex-col tw-justify-center tw-items-center" onclick="">
-          <i class="fas fa-bars"></i>
-          <span>Menu</span>
-        </div>
-      </div>
+
+      <a href="#" title="{{ 'Home'|t }}" rel="home" class="site-logo modal-open tw-self-stretch tw-flex tw-no-underline lg-tw-hidden tw- tw-border-r-2 tw-border-neutral-200 tw-justify-center">
+        <img class="tw-self-center tw-max-w-none tw-h-20" src="{{ crest_logo }}" alt="{{ 'Home'|t }}"/>
```
Comment (emerham) [inline templates/block/block--system-branding-block.html.twig]: OK so adding these help
tw-p-0.5 tw--ml-0.5

but it's still every so slightly off. If i set a padding-bottom: 3px or a padding-bottom: .2rem is even closer, but there is still the slightest jump. There is no built in tailwind class for that specific padding, so we either need to make it or just use an @apply in the css file.

**Review by emerham — CHANGES_REQUESTED** [review]

**Review by emerham — APPROVED** [review]
OK, Looks good to go

**Pull request merged**